hist

```diff
@@ -68,170 +137,290 @@ fail_unless_db_exists() {
 }
 
 cmd_config() {
-    echo "version:        ${VERSION}"
-    echo "source_history: ${HISTORIAN_SRC}"
-    echo "db:             ${HISTORIAN_DB}"
-    echo "sqlite3:        ${HISTORIAN_SQLITE3}"
+    cat <<EOF
+version:        ${VERSION}
+bash_history:   ${HISTORIAN_BASH}
+zsh_history:    ${HISTORIAN_ZSH}
+db:             ${HISTORIAN_DB}
+sqlite3:        ${HISTORIAN_SQLITE3}
+EOF
 }
 
 cmd_count() {
-    local args=$@
     preflight_check;
     ensure_db_exists;
 
-    ( cat <<SQL
+    historian_sqlite <<SQL
         SELECT COUNT(*) FROM history;
 SQL
-    ) | "${HISTORIAN_SQLITE3}" "${HISTORIAN_DB}";
 }
 
-cmd_import() {
-    local args=$@
-    preflight_check;
-    ensure_db_exists;
 
-    sanitized_src=$(mktemp)
-    cat ${HISTORIAN_SRC} \
-        | sed -e 's/"/'$(get_magic ${MAGIC_ENUM_QUOTE})'/g' \
-        > "${sanitized_src}" \
-    ;
-    if [ -n "${ZSH_EXTENDED_HISTORY}" ]; then
-        _import_zsh_extended_history;
-    else
-        _import_default;
-    fi
+csv_from_extended_zsh_history() {
+    # See: http://zsh.sourceforge.net/Doc/Release/Options.html#History
+    #
+    # NB. It is not 100% possible to disambiguate zsh lines from each other.
+    # This is because there can be any number of backslashes in the command history.
+    #
+    # However, in practice it is very unlikely for someone to construct
+    # lines like the following for zsh history.
+    # : 1234567890:0;whatever\
+    # : 1234567890:0;whatever\\\\
+    awk '
```

Comment:
I have mixed feelings about this much awk in a shell script, though I have already put in a decent amount of SQLite in here as well.